xcode/Subconscious/Shared/Services/AddressBookService.swift

```diff
@@ -34,77 +44,201 @@ class AddressBookService {
     private(set) var database: DatabaseService
     private var addressBook: [AddressBookEntry]?
     
+    private static let MAX_ATTEMPTS_TO_INCREMENT_PETNAME = 99
+    
+    static let logger = Logger(
+        subsystem: Config.default.rdns,
+        category: "AddressBookService"
+    )
+    
     init(noosphere: NoosphereService, database: DatabaseService) {
         self.noosphere = noosphere
         self.database = database
         self.addressBook = nil
     }
     
+    private func invalidateCache() {
+        addressBook = nil
+    }
+    
     /// Get the full list of entries in the address book.
     /// This is cached after the first use unless requested using refetch.
     /// If an error occurs producing the entries the resulting list will be empty.
-    func listEntries(refetch: Bool = false) -> AnyPublisher<[AddressBookEntry], Never> {
+    func listEntriesAsync(refetch: Bool = false) -> AnyPublisher<[AddressBookEntry], Error> {
```

Comment:
As discussed #501 introduces a conflicting change and we probably want to adopt the approach taken in that PR.